src/v2/wallet.js

```diff
@@ -1116,9 +1114,27 @@ Wallet.prototype.prebuildAndSignTransaction = function(params, callback) {
       throw error;
     }
 
-    // the prebuild can be overridden by providing an explicit tx
-    const txPrebuild = params.prebuildTx || (yield this.prebuildTransaction(params));
-    const userKeychain = yield this.baseCoin.keychains().get({ id: this._wallet.keys[0], reqId: params.reqId });
+    // call prebuildTransaction and keychains-get in parallel
+
+    // see if the user provided keychains in the params
+    const providedKeychains = params.verification && params.verification.keychains;
+
+    const response = yield Promise.props({
+      // the prebuild can be overridden by providing an explicit tx
+      txPrebuild: params.prebuildTx ? Promise.resolve(params.prebuildTx) : this.prebuildTransaction(params),
```

Comment:
This is a little too dense. Let's bring the conditionals out of here, and set up your promises before you pass them to `Promise.props()`